bin/zowe-setup-certificates.sh

```diff
@@ -0,0 +1,173 @@
+#!/bin/sh
+
+# Variables to be supplied:
+# - HOSTNAME - The hostname of the system running API Mediation
+# - IPADDRESS - The IP Address of the system running API Mediation
+# - VERIFY_CERTIFICATES - true/false - Should APIML verify certificates of services (defaults to true)
+# - ZOSMF_KEYRING - Name of the z/OSMF keyring
+# - ZOSMF_USER - z/OSMF server user ID
+# - EXTERNAL_CERTIFICATE - optional - Path to a PKCS12 keystore with a server certificate for APIML
+# - EXTERNAL_CERTIFICATE_ALIAS - optional - Alias of the certificate in the keystore
+# - EXTERNAL_CERTIFICATE_AUTHORITIES - optional - Public certificates of trusted CAs
+
+# - KEYSTORE_DIRECTORY - Location for generated certificates (defaults to /global/zowe/keystore)
+# - KEYSTORE_PASSWORD - a password that is used to secure EXTERNAL_CERTIFICATE keystore and 
+#                       that will be also used to secure newly generated keystores for API Mediation.
+# - ZOWE_USER_ID - zowe user id to set up ownership of the generated certificates
+
+
+# Set up logging
+LOG_DIR=${HOME}/zowe_certificate_setup_log
+# Make the log directory if needed - first time through - subsequent installs create new .log files
+if [[ ! -d $LOG_DIR ]]; then
+    mkdir -p $LOG_DIR
+    chmod a+rwx $LOG_DIR 
+fi
+
+export LOG_FILE="certificate_config_`date +%Y-%m-%d-%H-%M-%S`.log"
+LOG_FILE=$LOG_DIR/$LOG_FILE
+touch $LOG_FILE
+chmod a+rw $LOG_FILE
+
+echo "<zowe-setup-certificates.sh>" >> $LOG_FILE
+
+# process input parameters. 
+while getopts "p:" opt; do
+  case $opt in
+    p) CERTIFICATES_CONFIG_FILE=$OPTARG;;
+    \?)
+      echo "Invalid option: -$OPTARG" >&2
+      exit 1
+      ;;
+  esac
+done
+shift "$(($OPTIND-1))"
+
+export ZOWE_ROOT_DIR=$(cd $(dirname $0)/../;pwd)
```

Comment:
There is an issue when running this script from within the zowe-configure as $0 is zowe-configure not this script. I think the easiest way to do this might be to check if ZOWE_ROOT_DIR is already set and only run this code if it isn't?

Reply:
This sounds good. Thanks.